ExpSettingsVal.py

```diff
@@ -0,0 +1,331 @@
+'''
+ExpSettingsVal - 
+Validates Experimental Settings against a set of rules known to cause 
+the Compiler (Compiler.py) to fail if they are not followed
+
+Created on April 17, 2015
+
+Original Author: Brian Donovan
+
+Copyright 2015 Raytheon BBN Technologies
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+'''
+
+import h5py
+import Libraries
+import QGL.Channels
+
+from QGL.mm import multimethod
+from instruments.AWGs import APS, APS2, Tek5014
+
+from atom.api import Str
+
+channels = Libraries.channelLib.channelDict
+instruments = Libraries.instrumentLib.instrDict
+measurements = Libraries.measLib.filterDict
+sweeps = Libraries.sweepLib.sweepDict
+
+# The following naming conventions are currently enforced
+# See: https://github.com/BBN-Q/PyQLab/wiki
+#
+# Two LogicalMarkerChannels are required:
+#   1 digitizerTrig
+#   2 slaveTrig
+#
+# Logical Channels: 
+#   1 PhysicalChannel but be in library
+#	2 LogicalMarkerChannel must map to PhysicalMarkerChannel
+#
+# Physical Channels:
+#   1 PhysicalChannel must have an AWG assigned
+# 	2 The assigned AWG must exist in the library
+#	3 The name of the PhysicalChannel channel must be of the form AWGName-AWGChannel
+#   4 Device specific naming conventions
+#     APS: 12, 34, 1m1, 2m1, 3m1, 4m1
+#     APS2: 12, 12m1, 12m2, 12m3, 12m4
+#     Tek5014: 12, 34, 1m1, 1m2, 2m1, 2m2, 3m1, 3m2, 4m1, 4m2
+
+# Conventions to be added
+# 
+#
+
+#####################################################################################
+## Helper functions for list comprehension 
+
+def is_logicalmarker_channel(name):
+	return is_channel_type(name, QGL.Channels.LogicalMarkerChannel)	
+
+def is_physicalmarker_channel(name):
+	return is_channel_type(name, QGL.Channels.PhysicalMarkerChannel)	
+
+def is_physicalIQ_channel(name):
+	return is_channel_type(name, QGL.Channels.PhysicalQuadratureChannel)	
+
+def is_qubit_channel(name):
+	return is_channel_type(name, QGL.Channels.Qubit)	
+
+def is_measurement_channel(name):
+	return is_channel_type(name, QGL.Channels.Measurement)	
+
+def requires_physical_channel(name):
+	return (is_channel_type(name, QGL.Channels.Qubit) or
```

Comment:
This applies generically to *all* logical channels, so I think this can be simplified to
```python
return is_channel_type(name, QGL.Channels.LogicalChannel)
```